workbench-core/example/infrastructure/integration-tests/support/resources/datasets/dataset.ts

```diff
@@ -28,12 +31,48 @@ export default class Dataset extends Resource {
     this._api = `datasets/${params.id}`;
     this._clientSession = params.clientSession;
     this._children = new Map<string, Endpoint>();
+    this._permissions = new Map<string, IdentityPermission>();
+  }
+
+  public identityPermission(
+    identityPermission: Permission,
+    clientSession: ClientSession,
+    parentApi: string,
+    id: string
+  ): IdentityPermission {
+    return new IdentityPermission(identityPermission, clientSession, parentApi, id);
   }
 
   public endpoint(params: EndpointCreateParams): Endpoint {
     return new Endpoint(params);
   }
 
+  public async addAccess(requestBody: Record<string, unknown>): Promise<PermissionsResponse> {
+    const randomTextGenerator = new RandomTextGenerator(this._settings.get('runId'));
+    const response: AxiosResponse = await this._axiosInstance.post(`${this._api}/permissions`, requestBody);
+    const permissionsCreated: PermissionsResponse = response.data;
+
+    const permissions: Permission[] = permissionsCreated.data.permissions.map((p) => {
+      return {
+        action: p.accessLevel === 'read-only' ? 'READ' : 'UPDATE',
```

Comment:
should this be set via a switch that throws an error if an invalid value gets passed into to the method?  since `requestBody` isn't typed we can't guarantee we're getting valid data.

Reply:
validation added.